docs/sources/variables/variable-types/add-query-variable.md

```diff
@@ -8,6 +8,8 @@ weight = 100
 
 Query variables allow you to write a data source query that can return a list of metric names, tag values, or keys. For example, a query variable might return a list of server names, sensor IDs, or data centers. The variable values change as they dynamically fetch options with a data source query.
 
+Query variables are generally only supported for strings. If your query return numbers or any other data type, you may need to convert them to strings in order to use them as variables. For the Azure data source, for example, you can use the [tostring](https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/tostringfunction) function for this purpose.
```

Comment:
I would still omit the example or say something like "Check the query language documentation of your data source to learn how to transform the query result".

The example is not complete enough since it doesn't say how to use the function (and that will depend on the query) so I would skip it to avoid confusion.